### DISCUSSION TITLE
Build for Deploy/Kubernetes/Router failing

### DISCUSSION THREAD

**Discussion by vayu068** [Sunbird-Ed/Community#119] [Installation] [answered]
Build using release tags: release-4.6.0_RC4.
Please let me know the correct repository and Jenkinsfile to use as it seems it is incorrectly specified.

Started by user Pawan Verma
Rebuilds build #10
Checking out git https://github.com/project-sunbird/sunbird-devops.git into /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Router@script to read kubernetes/pipelines/deploy_core/Jenkinsfile
The recommended git tool is: NONE
No credentials specified
 > git rev-parse --resolve-git-dir /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Router@script/.git # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url https://github.com/project-sunbird/sunbird-devops.git # timeout=10
Using shallow fetch with depth 1
Fetching upstream changes from https://github.com/project-sunbird/sunbird-devops.git
 > git --version # timeout=10
 > git --version # 'git version 2.17.1'
 > git fetch --tags --progress --depth=1 -- https://github.com/project-sunbird/sunbird-devops.git +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git rev-parse origin/release-4.6.0_RC4^{commit} # timeout=10
 > git rev-parse release-4.6.0_RC4^{commit} # timeout=10
Checking out Revision 298c0e0cb57f5d8fa32bbc3676ae57c6c2127a6f (release-4.6.0_RC4)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 298c0e0cb57f5d8fa32bbc3676ae57c6c2127a6f # timeout=10
Commit message: "Issue #SB-28497 feat:updated enrollment list size env (#3180)"
 > git rev-list --no-walk 298c0e0cb57f5d8fa32bbc3676ae57c6c2127a6f # timeout=10
Running in Durability level: MAX_SURVIVABILITY
Loading library deploy-conf@heads/shared-lib
Attempting to resolve heads/shared-lib from remote references...
 > git --version # timeout=10
 > git --version # 'git version 2.17.1'
 > git ls-remote -- https://github.com/project-sunbird/sunbird-devops.git # timeout=10
The recommended git tool is: NONE
No credentials specified
 > git rev-parse --resolve-git-dir /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Router@libs/deploy-conf/.git # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url https://github.com/project-sunbird/sunbird-devops.git # timeout=10
Using shallow fetch with depth 1
Fetching upstream changes from https://github.com/project-sunbird/sunbird-devops.git
 > git --version # timeout=10
 > git --version # 'git version 2.17.1'
 > git fetch --no-tags --progress --depth=1 -- https://github.com/project-sunbird/sunbird-devops.git +refs/heads/*:refs/remotes/origin/* refs/heads/shared-lib # timeout=10
Checking out Revision 778da7ce7f97efd003eed98071921a837d209343 (778da7ce7f97efd003eed98071921a837d209343)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 778da7ce7f97efd003eed98071921a837d209343 # timeout=10
Commit message: "fix: move variables to be accessible across both try blocks"
 > git rev-list --no-walk 298c0e0cb57f5d8fa32bbc3676ae57c6c2127a6f # timeout=10
First time build. Skipping changelog.
[Pipeline] Start of Pipeline
[Pipeline] node
Running on Jenkins in /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Router
[Pipeline] {
[Pipeline] stage
[Pipeline] { (checkout public repo)
[Pipeline] echo
Found .git folder. Clearing it..
[Pipeline] sh
+ git clean -fxd
Removing metadata.json
Removing router@tmp/
[Pipeline] checkout
The recommended git tool is: NONE
No credentials specified
 > git rev-parse --resolve-git-dir /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Router/.git # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url https://github.com/project-sunbird/sunbird-devops.git # timeout=10
Using shallow fetch with depth 1
Fetching upstream changes from https://github.com/project-sunbird/sunbird-devops.git
 > git --version # timeout=10
 > git --version # 'git version 2.17.1'
 > git fetch --tags --progress --depth=1 -- https://github.com/project-sunbird/sunbird-devops.git +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git rev-parse origin/release-4.6.0_RC4^{commit} # timeout=10
 > git rev-parse release-4.6.0_RC4^{commit} # timeout=10
Checking out Revision 298c0e0cb57f5d8fa32bbc3676ae57c6c2127a6f (release-4.6.0_RC4)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 298c0e0cb57f5d8fa32bbc3676ae57c6c2127a6f # timeout=10
Commit message: "Issue #SB-28497 feat:updated enrollment list size env (#3180)"
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (deploy)
[Pipeline] ansiColor
[Pipeline] {

[Pipeline] sh
+ echo Deploy/dev/Kubernetes/Router
[Pipeline] sh
+ echo Deploy/dev/Kubernetes/Router
[Pipeline] sh
+ echo Deploy/dev/Kubernetes/Router
[Pipeline] stage
[Pipeline] { (parameter checks)
[Pipeline] echo
[1m[32mFound environment variable named hub_org with value as: vayu068[0m
[Pipeline] copyArtifacts
Copied 1 artifact from "ArtifactUpload » dev » Core » Router" build number 3
[Pipeline] sh
+ jq -r .image_name metadata.json
[Pipeline] sh
+ jq -r .node_name metadata.json
[Pipeline] echo
[1m[33mimage_tag not specified, using the image_tag specified in metadata.json. [0m
[Pipeline] sh
+ jq -r .image_tag metadata.json
[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] sh
+ echo Deploy/dev/Kubernetes/Router
[Pipeline] sh
+ pwd
[Pipeline] ansiColor
[Pipeline] {

[Pipeline] stage
[Pipeline] { (ansible-run)
[Pipeline] echo
{env=dev, module=Kubernetes, jobName=Router, absolute_job_path=ArtifactUpload/dev/Core/Router, agent=master, image_name=router, image_tag=release-3.8.0_RC2_2_bot, currentWs=/var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Router, ansiblePlaybook=/var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Router/kubernetes/ansible/deploy_core_service.yml, ansibleExtraArgs=--extra-vars "chart_path=/var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Router/kubernetes/helm_charts/core/router release_name=router image_tag=release-3.8.0_RC2_2_bot image_name=router role_name=sunbird-deploy" --vault-password-file /var/lib/jenkins/secrets/vault-pass}
[Pipeline] dir
Running in /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Router/private
[Pipeline] {
[Pipeline] echo
[1m[33mInfo: Branch override is enabled[0m
[Pipeline] checkout
The recommended git tool is: NONE
using credential github-private-repo
Cloning the remote Git repository
Using shallow clone with depth 1
Avoid fetching tags
Cloning repository https://github.com/vayu068/Sunbird_Private_Azure.git
 > git init /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Router/private # timeout=10
Fetching upstream changes from https://github.com/vayu068/Sunbird_Private_Azure.git
 > git --version # timeout=10
 > git --version # 'git version 2.17.1'
using GIT_ASKPASS to set credentials 
 > git fetch --no-tags --progress --depth=1 -- https://github.com/vayu068/Sunbird_Private_Azure.git +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git config remote.origin.url https://github.com/vayu068/Sunbird_Private_Azure.git # timeout=10
 > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
Avoid second fetch
 > git rev-parse origin/master^{commit} # timeout=10
Checking out Revision 2d132221ea376535644b4f6d2718e6dc8e2bfd74 (origin/master)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 2d132221ea376535644b4f6d2718e6dc8e2bfd74 # timeout=10
Commit message: "Update common.yml"
 > git rev-list --no-walk 2d132221ea376535644b4f6d2718e6dc8e2bfd74 # timeout=10
[Pipeline] }
[Pipeline] // dir
[Pipeline] sh
+ rsync -Lkr /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Router/private/ansible/inventory/dev/Kubernetes/common.yml /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Router/private/ansible/inventory/dev/Kubernetes/hosts /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Router/private/ansible/inventory/dev/Kubernetes/secrets.yml /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Router/ansible/inventory/env/
+ [ -f /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Router/ansible/inventory/env/kubernetes.yaml ]
+ ansible-playbook -i /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Router/ansible/inventory/env/hosts /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Router/kubernetes/ansible/deploy_core_service.yml --extra-vars chart_path=/var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Router/kubernetes/helm_charts/core/router release_name=router image_tag=release-3.8.0_RC2_2_bot image_name=router role_name=sunbird-deploy --vault-password-file /var/lib/jenkins/secrets/vault-pass
[1;35m[WARNING]: Invalid characters were found in group names but not replaced, use[0m
[1;35m-vvvv to see details[0m
[1;35m[0m
[1;35m[WARNING]:  * Failed to parse /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/[0m
[1;35mRouter/ansible/inventory/env/hosts with yaml plugin: Syntax Error while loading[0m
[1;35mYAML.   found unexpected ':'  The error appears to be in '/var/lib/jenkins/work[0m
[1;35mspace/Deploy/dev/Kubernetes/Router/ansible/inventory/env/hosts': line 1, column[0m
[1;35m5, but may be elsewhere in the file depending on the exact syntax problem.  The[0m
[1;35moffending line appears to be:   [all:vars]     ^ here[0m
[1;35m[0m
[1;35m[WARNING]:  * Failed to parse /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/[0m
[1;35mRouter/ansible/inventory/env/hosts with ini plugin: /var/lib/jenkins/workspace/[0m
[1;35mDeploy/dev/Kubernetes/Router/ansible/inventory/env/hosts:143: Section[0m
[1;35m[core:children] includes undefined group: postgresql-slave[0m
[1;35m[0m
[1;35m[WARNING]: Unable to parse /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Rou[0m
[1;35mter/ansible/inventory/env/hosts as an inventory source[0m
[1;35m[0m
[1;35m[WARNING]: No inventory was parsed, only implicit localhost is available[0m
[1;35m[0m
[1;35m[WARNING]: provided hosts list is empty, only localhost is available. Note that[0m
[1;35mthe implicit localhost does not match 'all'[0m
[1;35m[0m
[1;35m[WARNING]: While constructing a mapping from /var/lib/jenkins/workspace/Deploy/[0m
[1;35mdev/Kubernetes/Router/kubernetes/ansible/roles/sunbird-[0m
[1;35mdeploy/defaults/main.yml, line 3, column 1, found a duplicate dict key[0m
[1;35m(sunbird_time_zone). Using last defined value only.[0m
[1;35m[0m
[1;35m[WARNING]: While constructing a mapping from /var/lib/jenkins/workspace/Deploy/[0m
[1;35mdev/Kubernetes/Router/kubernetes/ansible/roles/sunbird-[0m
[1;35mdeploy/defaults/main.yml, line 3, column 1, found a duplicate dict key[0m
[1;35m(nginx_per_ip_connection_limit). Using last defined value only.[0m
[1;35m[0m
[1;35m[WARNING]: While constructing a mapping from /var/lib/jenkins/workspace/Deploy/[0m
[1;35mdev/Kubernetes/Router/kubernetes/ansible/roles/sunbird-[0m
[1;35mdeploy/defaults/main.yml, line 3, column 1, found a duplicate dict key[0m
[1;35m(sunbird_portal_player_cdn_enabled). Using last defined value only.[0m
[1;35m[0m
[1;35m[WARNING]: While constructing a mapping from /var/lib/jenkins/workspace/Deploy/[0m
[1;35mdev/Kubernetes/Router/kubernetes/ansible/roles/sunbird-[0m
[1;35mdeploy/defaults/main.yml, line 3, column 1, found a duplicate dict key[0m
[1;35m(dial_liveness_readiness). Using last defined value only.[0m
[1;35m[0m
[1;35m[WARNING]: While constructing a mapping from /var/lib/jenkins/workspace/Deploy/[0m
[1;35mdev/Kubernetes/Router/kubernetes/ansible/roles/sunbird-[0m
[1;35mdeploy/defaults/main.yml, line 928, column 5, found a duplicate dict key[0m
[1;35m(failureThreshold). Using last defined value only.[0m
[1;35m[0m
[1;35m[WARNING]: While constructing a mapping from /var/lib/jenkins/workspace/Deploy/[0m
[1;35mdev/Kubernetes/Router/kubernetes/ansible/roles/sunbird-[0m
[1;35mdeploy/defaults/main.yml, line 938, column 5, found a duplicate dict key[0m
[1;35m(failureThreshold). Using last defined value only.[0m
[1;35m[0m
[1;35m[WARNING]: While constructing a mapping from /var/lib/jenkins/workspace/Deploy/[0m
[1;35mdev/Kubernetes/Router/kubernetes/ansible/roles/sunbird-[0m
[1;35mdeploy/defaults/main.yml, line 951, column 5, found a duplicate dict key[0m
[1;35m(failureThreshold). Using last defined value only.[0m
[1;35m[0m
[1;35m[WARNING]: While constructing a mapping from /var/lib/jenkins/workspace/Deploy/[0m
[1;35mdev/Kubernetes/Router/kubernetes/ansible/roles/sunbird-[0m
[1;35mdeploy/defaults/main.yml, line 961, column 5, found a duplicate dict key[0m
[1;35m(failureThreshold). Using last defined value only.[0m
[1;35m[0m

PLAY [local] *******************************************************************
[1;35m[WARNING]: While constructing a mapping from /var/lib/jenkins/workspace/Deploy/[0m
[1;35mdev/Kubernetes/Router/ansible/inventory/env/group_vars/all.yml, line 4, column[0m
[1;35m1, found a duplicate dict key (app_es_snapshot_host). Using last defined value[0m
[1;35monly.[0m
[1;35m[0m
[1;35m[WARNING]: While constructing a mapping from /var/lib/jenkins/workspace/Deploy/[0m
[1;35mdev/Kubernetes/Router/ansible/inventory/env/group_vars/all.yml, line 4, column[0m
[1;35m1, found a duplicate dict key (sunbird_es_host). Using last defined value only.[0m
[1;35m[0m
[1;35m[WARNING]: While constructing a mapping from /var/lib/jenkins/workspace/Deploy/[0m
[1;35mdev/Kubernetes/Router/ansible/inventory/env/group_vars/all.yml, line 4, column[0m
[1;35m1, found a duplicate dict key (log_es_snapshot_host). Using last defined value[0m
[1;35monly.[0m
[1;35m[0m

TASK [check for kong db migration] *********************************************
[0;36mskipping: [localhost][0m

TASK [sunbird-deploy : template values.yaml file] ******************************
[0;31mfatal: [localhost]: FAILED! => {[0m
[0;31m    "changed": false[0m
[0;31m}[0m
[0;31m[0m
[0;31mMSG:[0m
[0;31m[0m
[0;31mCould not find or access '/var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Router/kubernetes/helm_charts/core/router/values.j2' on the Ansible Controller.[0m
[0;31mIf you are using a module and expect the file to exist on the remote, see the remote_src option[0m
[0;31m[0m

PLAY RECAP *********************************************************************
[0;31mlocalhost[0m                  : ok=0    changed=0    unreachable=0    [0;31mfailed=1   [0m [0;36mskipped=1   [0m rescued=0    ignored=0   

[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] }
[Pipeline] // stage
[Pipeline] ansiColor
[Pipeline] {

[Pipeline] stage
[Pipeline] { (slack_notify)
[Pipeline] sh
+ echo Deploy/dev/Kubernetes/Router
[Pipeline] echo
[33m[1mCould not find env specific Slack channel. Check for global slack channel..[0m
[Pipeline] echo
[33m[1mCould not find slack environment variable. Skipping slack notification..[0m
[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] ansiColor
[Pipeline] {

[Pipeline] stage
[Pipeline] { (email_notify)
[Pipeline] sh
+ echo Deploy/dev/Kubernetes/Router
[Pipeline] echo
[33m[1mCould not find env specific email group. Check for global email group..[0m
[Pipeline] echo
[33m[1mCould not find global email group variable. Skipping email notification..[0m
[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
ERROR: script returned exit code 2
Finished: FAILURE

**Comment by beepdot** [comment, marked as answer]
The deployment chart is present in this repo - https://github.com/project-sunbird/sunbird-bot.git
Please use this repo in jenkins job configuration and tag as release-3.9.0_RC1

**Reply by vayu068** [reply]
> https://github.com/project-sunbird/sunbird-bot.git

Name of JenkinsFile is chatbot.Jenkinsfile ?? Please confirm

**Reply by vayu068** [reply]
This works. Thanks

**Reply by vayu068** [reply]
Role name used chatbot
